pkg/variantregistry/ocp.go

```diff
@@ -0,0 +1,448 @@
+package variantregistry
+
+import (
+	"context"
+	_ "embed"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/bigquery"
+	"cloud.google.com/go/storage"
+	"github.com/hashicorp/go-version"
+	"github.com/openshift/sippy/pkg/dataloader/prowloader"
+	"github.com/openshift/sippy/pkg/dataloader/prowloader/gcs"
+	"github.com/openshift/sippy/pkg/testidentification"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/api/iterator"
+)
+
+// OCPVariantLoader generates a mapping of job names to their variant map for all known jobs.
+type OCPVariantLoader struct {
+	BigQueryClient *bigquery.Client
+	VariantManager testidentification.VariantManager
+	bkt            *storage.BucketHandle
+}
+
+func NewOCPVariantLoader(
+	bigQueryClient *bigquery.Client,
+	gcsClient *storage.Client,
+	gcsBucket string) *OCPVariantLoader {
+
+	bkt := gcsClient.Bucket(gcsBucket)
+	return &OCPVariantLoader{
+		BigQueryClient: bigQueryClient,
+		bkt:            bkt,
+	}
+
+}
+
+type prowJobLastRun struct {
+	JobName  string              `bigquery:"prowjob_job_name"`
+	JobRunID string              `bigquery:"prowjob_build_id"`
+	URL      bigquery.NullString `bigquery:"prowjob_url"`
+}
+
+// LoadExpectedJobVariants queries all known jobs from the gce-devel "jobs" table (actually contains job runs).
+// This effectively is every job that actually ran in the last several years.
+func (v *OCPVariantLoader) LoadExpectedJobVariants(ctx context.Context) (map[string]map[string]string, error) {
```

Comment:
This may cause problems for our planned view representing these variants as columns for integration and ease of use, I don't think that's going to work with multi-valued variants, and would require a more involved integration into the three apps using this thing. Maybe that's worthwhile but we need to make sure.

My hope was that if it's important enough we need to slice on it, but not important enough to get a variant on all jobs ,that it would collapse into a pre-existing variant with hyphenation, i.e. virtualmedia could be something like: Installer=ipi-virtualmedia. 

Need to think on this and maybe discuss, I'm not sure which way to go.

Reply:
Team discussed, we think there's a good chance we can get by with the current implementation with a few levers. Some of the desired "tags" we discussed fit into existing variants such as suite, others are probably worthy of getting a variant on all jobs, and those that alter an existing variant significantly we can hyphenate. We'll see how it goes.